Caboodle/Sms/Sms.ios.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Threading.Tasks;
+using MessageUI;
+
+namespace Microsoft.Caboodle
+{
+    public static partial class Sms
+    {
+        public static bool IsComposeSupported
+            => MFMessageComposeViewController.CanSendText;
+
+        public static Task ComposeAsync(SmsMessage message)
+        {
+            if (message == null)
+                throw new ArgumentNullException(nameof(message));
+            message.Validate();
+
+            // do this first so we can throw as early as possible
+            var controller = Platform.GetCurrentViewController();
+
+            // show the controller
+            var tcs = new TaskCompletionSource<bool>();
+            var messageController = new MFMessageComposeViewController
+            {
+                Body = message.Body,
+                Recipients = new[] { message.Recipient }
+            };
+            messageController.Finished += (sender, e) =>
```

Comment:
Does this get triggered no matter what? even if it is canceled? Is it better ot fire and forget?

Reply:
This appears to be the case. The docs do say we have to explicitly hide the controller in this event and we are provided a result (success, canceled, failed)